src/Algorithms/DirectSearch/CMA.cpp

```diff
@@ -37,9 +37,21 @@
 #include <shark/Core/Exception.h>
 #include <shark/Algorithms/DirectSearch/Operators/Evaluation/PenalizingEvaluator.h>
 #include <shark/Algorithms/DirectSearch/Operators/Selection/ElitistSelection.h>
+#include <shark/Core/utility/KeyValuePair.h>
 
 using namespace shark;
 
+namespace{
+	//computes percentile({|1-R|,|2-R|,...,|N-R|},q)
+	int deltaLim(int rank, std::size_t N, double percentile){
+		blas::vector<int> rankDistr(N);
+		std::iota(rankDistr.begin(),rankDistr.end(),1-rank);
+		noalias(rankDistr) = abs(rankDistr);
+		std::sort(rankDistr.begin(),rankDistr.end());
+		return rankDistr[std::size_t(percentile * (rankDistr.size()-1))];
```

Comment:
Maybe consider using std::nth_element instead to reduce the complexity?
Well, I know, this code is not a performance bottleneck.